samples/dashboards/demo/personal-portfolio/demo.js

```diff
@@ -0,0 +1,590 @@
+            key,
+            values.map((value, i) => [dates[i], value])
+        ])
+    );
+
+    const holdings = [],
+        investedAmounts = [],
+        dataGridData = [];
+
+    stockCollection.forEach(stock => {
+        const { holding, investedAmount } = generatePortfolio(
+            basicInvestmentPlan,
+            processedData[stock.SecID]
+        );
+
+        holdings.push(holding);
+        investedAmounts.push(investedAmount);
+
+    });
+
+    const investedAmountTotal = getCurrentTotal(investedAmounts),
+        lastHoldingTotal = getCurrentTotal(holdings),
+        annualInvestment = 200 * 12 * holdings.length;
+
+    // Generate columns for the datagrid
+    stockCollection.forEach((stock, i) => {
+        const len = holdings[i].length,
+            lastHolding = holdings[i][len - 1],
+            ISIN = stock.ISIN,
+            tradingSymbol = stock.tradingSymbol;
+
+        dataGridData.push([
+            tradingSymbol,
+            ISIN,
+            Math.round(lastHolding / lastHoldingTotal * 100)
+        ]);
+    });
+
+    // Generate portfolio for risk score
+    const portfolio = {
+        name: 'PersonalPortfolio',
+        currency: 'EUR',
+        totalValue: lastHoldingTotal,
+        // Holdings with equal weights
+        holdings: getHoldings(100 / stockCollection.length)
+    };
+
+    // Creating chart and KPI options
+    const walletChartOptions = {
+        plotOptions: {
+            series: {
+                marker: {
+                    enabled: false
+                }
+            }
+        },
+        chart: {
+            height: 400
+        },
+        accessibility: {
+            description: 'displaying portfolio performance and investments'
+        },
+        rangeSelector: {
+            animate: false,
+            x: 0,
+            y: 0,
+            inputEnabled: false,
+            dropdown: 'never',
+            selected: 4
+        },
+        navigator: {
+            enabled: false
+        },
+        title: {
+            text: 'Portfolio performance'
+        },
+        yAxis: {
+            accessibility: {
+                description: 'price in euro'
+            }
+        },
+        tooltip: {
+            shared: true,
+            xDateFormat: '%Y-%m-%d',
+            formatter: function () {
+                const holdVal = this.points[0].y; // Value of Series 1
+                const investedVal = this.points[1].y; // Value of Series 2
+                const percentageDiff = (
+                    (holdVal - investedVal) / investedVal * 100).toFixed(2);
+
+                return `<b>${Highcharts.dateFormat(
+                    '%e - %b - %Y',
+                    new Date(this.x)
+                )}</b><br/>
+                        Invested: €${investedVal.toFixed(2)}<br/>
+                        Holding: €${holdVal.toFixed(2)}<br/>
+                        Yield: ${percentageDiff}%`;
+            }
```

Comment:
Could you try rewriting this using a [format](https://www.highcharts.com/docs/chart-concepts/templating) option?

Reply:
Regarding the responsiveness, it works. The issue happened when having other panels open in utils.